src/dbnode/encoding/m3tsz/encoder.go

```diff
@@ -201,16 +204,18 @@ func (enc *encoder) writeTimeUnit(tu xtime.Unit) bool {
 	return true
 }
 
-func (enc *encoder) writeFirstTime(t time.Time, ant ts.Annotation, tu xtime.Unit) error {
+// WriteFirstTime encodes the first timestamp.
+func (enc *Encoder) WriteFirstTime(t time.Time, ant ts.Annotation, tu xtime.Unit) error {
 	// NB(xichen): Always write the first time in nanoseconds because we don't know
 	// if the start time is going to be a multiple of the time unit provided.
 	nt := xtime.ToNormalizedTime(enc.t, time.Nanosecond)
 	enc.os.WriteBits(uint64(nt), 64)
-	return enc.writeNextTime(t, ant, tu)
+	enc.writeAnnotation(ant)
+	return enc.WriteNextTime(t, ant, tu)
 }
 
-func (enc *encoder) writeNextTime(t time.Time, ant ts.Annotation, tu xtime.Unit) error {
-	enc.writeAnnotation(ant)
+// WriteFirstTime encodes the next (non-first) timestamp.
+func (enc *Encoder) WriteNextTime(t time.Time, ant ts.Annotation, tu xtime.Unit) error {
```

Comment:
The comment is wrong, I'll fix that. Not sure what you mean by only encoding the annotation once?

Reply:
Hm yeah, is there a reason that `enc.writeAnnotation(ant)` is not called in `WriteNextTime` anymore?

Reply:
I think I made this change as some other refactoring I did with an earlier implementation. The way I had it was still correct since I added a call to `writeAnnotation()` in `writeNext()` before the call to `writeNextTimestamp()` but agree it was confusing. Set it back to behave how it used to